### ISSUE TITLE
git: Fails to re-create working directory

### ISSUE THREAD

**Issue by ReillyBrogan** [serpent-os/boulder#58]
To reproduce:
1. Build glibc (until merged use the one in PR 5).
2. Delete the `sourceware.org` directory under the git upstreams (`sudo rm -rf /var/cache/boulder/upstreams/git/sourceware.org`)
3. Build glibc again

This results in the following error:
```
sudo boulder bi stone.yml -p local-x86_64
[10:21:36] INFO      Stage success: clean-root
[10:21:36] INFO      Stage success: create-root
[10:21:36] ERROR     Exception: Failed to open working directory (No such file or directory)
[10:21:36] ERROR     Stage failure: fetch-upstreams
```

**Comment by ReillyBrogan** [comment]
I noticed that you need to delete `/var/cache/boulder/upstreams/staging/git/sourceware.org/` as well but this error should still be handled by Boulder.

**Comment by GZGavinZhao** [comment]
I'll take a look at the two git issues this weeekend.

**Comment by GZGavinZhao** [comment]
Just tested, this was also fixed with #72.
